pyteal/ast/abi/tuple.py

```diff
@@ -381,7 +381,9 @@ def __init__(
         value2_type_spec: TypeSpec,
         value3_type_spec: TypeSpec,
     ) -> None:
-        super().__init__(value1_type_spec, value2_type_spec, value3_type_spec)
+        super().__init__(
+            TupleTypeSpec(value1_type_spec, value2_type_spec, value3_type_spec)
```

Comment:
I'm afraid for consistency, these would have to accept a single `TupleTypeSpec` as an arg and error if the number of values in that type spec is different than what this class should expect.